apps/jobs/serializers.py

```diff
@@ -62,6 +66,29 @@ class Meta:
             "is_verified_by_host",
         )
 
+    # Determines whether environment error file should be visible to caller.
+    def is_code_upload_environment_log_file_visible(self, context):
+        if not context:
+            return False
+        curr_user = context.get("request").user
+        if not curr_user:
+            return False
+        challenge_phase = self.fields.get("challenge_phase")
+        if not challenge_phase:
+            return False
+        challenge_phase_obj = list(challenge_phase.get_queryset())
+        if not challenge_phase_obj:
+            return False
+        challenge = challenge_phase_obj[0].challenge
+        if not challenge:
+            return False
+        challenge_host_team = challenge.creator
+        if not challenge_host_team:
+            return False
+        challenge_hosts_pk = challenge_host_team.get_all_challenge_hosts_pk()
```

Comment:
Can you refactor this check to be more readable.

Reply:
I have now refactored the check.